src/IO/H5/VolumeData.cpp

```diff
@@ -134,61 +215,89 @@ double VolumeData::get_observation_value(const size_t observation_id) const
                                           "observation_value");
 }
 
-std::vector<std::string> VolumeData::list_grids(
-    const size_t observation_id) const noexcept {
-  detail::OpenGroup observation_group(
-      volume_data_group_.id(),
-      "ObservationId" + std::to_string(observation_id), AccessType::ReadOnly);
-  return get_group_names(observation_group.id(), "");
-}
-
 std::vector<std::string> VolumeData::list_tensor_components(
-    size_t observation_id, const std::string& grid_name) const noexcept {
-  detail::OpenGroup spatial_group(
-      volume_data_group_.id(),
-      "ObservationId" + std::to_string(observation_id) + "/" + grid_name,
-      AccessType::ReadOnly);
-  auto tensor_components = get_group_names(spatial_group.id(), "");
+    const size_t observation_id) const noexcept {
+  auto tensor_components =
+      get_group_names(volume_data_group_.id(),
+                      "ObservationId" + std::to_string(observation_id));
   std::remove(tensor_components.begin(), tensor_components.end(),
               "connectivity");
   tensor_components.pop_back();
+  std::remove(tensor_components.begin(), tensor_components.end(),
+              "total_extents");
+  tensor_components.pop_back();
+  std::remove(tensor_components.begin(), tensor_components.end(), "grid_names");
+  tensor_components.pop_back();
+
+  // std::remove moves the element to the end of the vector, so we still need to
+  // actually erase it from the vector
   return tensor_components;
 }
+std::vector<std::string> VolumeData::get_grid_names(size_t observation_id) const
+    noexcept {
+  const std::string path = "ObservationId" + std::to_string(observation_id);
+  detail::OpenGroup observation_group(volume_data_group_.id(), path,
+                                      AccessType::ReadOnly);
+  const std::vector<char> names =
+      h5::read_data<1, std::vector<char>>(observation_group.id(), "grid_names");
+  const std::string all_names(names.begin(), names.end());
+  std::vector<std::string> grid_names{};
+  boost::split(grid_names, all_names, [](char c) noexcept { return c == ':'; });
```

Comment:
`':'` -> `separator()` function (you'll need to capture `this` possibly
`char c` -> `const char c`